nixos/modules/services/desktops/system76-scheduler.nix

```diff
@@ -0,0 +1,310 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.system76-scheduler;
+
+  inherit (builtins) concatStringsSep map toString attrNames;
+  inherit (lib) boolToString types mkOption literalExpression mdDoc optional mkIf mkMerge;
+  inherit (types) nullOr listOf bool int ints float str enum;
+
+  latencyProfile = { latency
```

Comment:
I don't know why, but this code is hopelessly ugly.

Reply:
It definitely is, but why "hopelessly"?  Is it the overall approach, my lacking FP chops, the Nix language limitations, all of the above?

Reply:
Any better now?